### ISSUE TITLE
The PSI-Mass Spectrometry ontology hierarchy does not look right in OLS website

### ISSUE THREAD

**Issue by javizca** [EBISPOT/OLS#623]
Dear colleagues,

https://www.ebi.ac.uk/ols/ontologies/ms

It seems that something is not right because in the first hierarchical level of the ontology, many terms from the ontology (which should not) appear and also the complete UNIT ontology as well. This is not right. I don't know whether it is the fault of the parser or we have formatted wrongly the OWL file in any of the recent changes.

Could you please help?

Thanks,

Juan

**Comment by henrietteharmse** [comment]
I think there are multiple issues here:

1.  I see the following in the ms.owl file: `<owl:Class rdf:about="http://purl.obolibrary.org/obo/xsd_positiveInteger"/>`. This to the best of my knowledge this is not the standard way to create an ontology. 
2. The ontology is inconsistent, which can also be an indication that the way the ontology is created may be problematic. 

In both cases 1 and 2 I strongly suggest the creators of the ontology to get in touch with @dosumis to discuss ways to address this.

3.  The ontology is making use of punning, i.e., "acre" (UO_0010025) is both a class and an individual. This may be due to an issue in OLS, but this is something that is still under investigation.

**Comment by henrietteharmse** [comment]
OLS has been updated to fix (3) above. (1) and (2) still needs to be addressed by the designers of the PSIMass Spectrometry ontology.

**Comment by javizca** [comment]
No, the issues are not solved still. Work in progress.

**Comment by henrietteharmse** [comment]
Though we can update MS on OLS, it will still be broken on OLS while these are still an issue.

**Comment by edeutsch** [comment]
Okay, thank you, @dosumis how can we address these issues?

**Comment by dosumis** [comment]
HI @edeutsch - Maybe a quick call to talk through the issues would be the most efficient way to proceed?  If you're interested, we could help you switch to a build pipeline with QC to prevent issues like this from occurring.

**Comment by matentzn** [comment]
3. is fine.. Class-individual is legal punning (I would not generally recommend it though)

I don't see an OLS issue. OLS uses the MS PURL http://purl.obolibrary.org/obo/ms.owl to download the ontology. This ontology imports the entirety the PATO and STATO. This is why your top level looks so "untidy":

<img width="597" alt="image" src="https://user-images.githubusercontent.com/7070631/192465562-4d43bdbf-9cef-4c83-a060-26488cdae275.png">

There are many unsats as @henrietteharmse says:

<img width="291" alt="image" src="https://user-images.githubusercontent.com/7070631/192464770-11e9fb52-cd5e-43ad-b94e-93b750c9b224.png">

Open MS in Protege and debug it. Or use [ROBOT explain](robot.obolibrary.org/explain):

```
robot explain -i ms.owl --reasoner ELK -M unsatisfiability --unsatisfiable random:10 --explanation unsat_1.md
```

This will help you identify the logical errors! Hope that helps!

**Comment by dosumis** [comment]
> It seems that something is not right because in the first hierarchical level of the ontology, many terms from the ontology (which should not) appear and also the complete UNIT ontology as well.

To improve the view in OLS, you can add preferred roots to your OLS config. This will set the terms that are visible at the top level by default.  @serjoshua should be able to point you to documentation on how to do this.

**Comment by henrietteharmse** [comment]
@dosumis  Preferred roots will not sort out issues like the `xsd_anyURI` etc being displayed at the root. That should be resolved in the ontology? 

![image](https://user-images.githubusercontent.com/32327979/192469407-7b13bbe6-9ddd-4844-8eec-be476ffcb98b.png)

**Comment by matentzn** [comment]
`obo:xsd_anyUri` is such a weird way to talk about xsd:anyUri datatype.. that said if the preferred root is some other class somewhere, no other classes will be shown regardless of if they are floating to the top, right? https://www.ebi.ac.uk/ols/ontologies/mondo has lots of floating classes but looks all nice and tidy..

**Comment by henrietteharmse** [comment]
@matentzn Agreed about `obo:xsd_anyUri`. That is why I suggested they get in touch with your teams.
